Ports/build_all.sh

```diff
@@ -1,45 +1,62 @@
 #!/usr/bin/env bash
 
+build=true
 clean=false
 verbose=false
 
-case "$1" in
-    clean)
-        clean=true
-        ;;
-    verbose)
-        verbose=true
-        ;;
-    *)
-        ;;
-esac
-
-case "$2" in
-    clean)
-        clean=true
-        ;;
-    verbose)
-        verbose=true
-        ;;
-    *)
-        ;;
-esac
+for argument in "$@"; do
+    case "$argument" in
+        clean)
+            clean=true
+            ;;
+        clean_only)
+            build=false
+            clean=true
+            ;;
+        verbose)
+            verbose=true
+            ;;
+        *)
+            ;;
+    esac
+done
 
 some_failed=false
 built_ports=""
 
-for file in *; do
-    if [ -d $file ]; then
-        pushd $file > /dev/null
-            port=$(basename $file)
+if [ "$clean" == true ]; then
+    for file in *; do
+        if [ -d "$file" ]; then
```

Comment:
Could probably just early-continue here instead of wrapping everything in an if. I know that this doesn't follow the same style as the other loop, but I'm hoping to eventually change that as well if I get the chance, just to keep it from getting further and further indented.